### PR TITLE
Expose hearingSummary.hearingDay.hasSharedResults flag

### PR DESCRIPTION
**Pull request by laurentqro**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-718)

CDA will need to expose the status of results shared for VCD to consume during the search operation. This information will be available within the hearingSummary.hearingDay.hasResultsShared  as part of HMCTS’ search API response.  More detail can be found on this link

## Why

This information will be displayed on the VCD screen which will enable billing caseworkers to understand the status of the results shared. 

## Definition of Done

* [x] Results shared status is made available within CDA’s search API & tested.
* [x] The relevant API docs are updated to reflect this new info. 
* [x] v1/v2 services are updated.

### CODE REVIEW

**Review by chris-groves — COMMENTED** [review]
Looks good. 

Is it worth adding tests for the hearing_summary model and serializer?

The has_shared_results field renders between the hearing_type and hearing_days fields, whereas the swagger documentation states that has_shared_results will come after hearing_type and hearing_days.

**Comment by arthurashman** [comment]
The docs say that hasSharedResults is on the HeargingSummary>HearingDay, not the HearingSummary

File: app/models/hearing_summary.rb [hunk 1]
```diff
@@ -39,8 +39,6 @@ def court_centre
     HmctsCommonPlatform::CourtCentre.new(body["courtCentre"])
   end
 
-private
-
   def hearing_days
```
Comment (arthurashman) [inline app/models/hearing_summary.rb:42]: Can we have a test for this?

File: app/models/hearing_summary.rb [hunk 2]
```diff
@@ -39,8 +39,6 @@ def court_centre
     HmctsCommonPlatform::CourtCentre.new(body["courtCentre"])
   end
```
Comment (arthurashman) [inline app/models/hearing_summary.rb:41]: models/hearing_summary.rb has a resulted? method. Wondering if the logic for calculating this is faulty/outdated, and we could now use the hasSharedResults attribute?

**Review by arthurashman — APPROVED** [review]
Looks good. Thank you for doing the refactor work!

**Pull request merged**